core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -176,13 +176,6 @@ private[spark] class HeartbeatReceiver(sc: SparkContext, clock: Clock)
     Option(self).map(_.ask[Boolean](ExecutorRegistered(executorId)))
   }
 
-  /**
-   * If the heartbeat receiver is not stopped, notify it of executor registrations.
-   */
-  override def onExecutorAdded(executorAdded: SparkListenerExecutorAdded): Unit = {
-    addExecutor(executorAdded.executorId)
-  }
-
```

Comment:
It looks inconsistent to have scheduler call into heartbeat receiver for execution addition and not removal.